src/System.Runtime/tests/System/Char.cs

```diff
@@ -894,44 +866,58 @@ public static void TestToUpperInvariant()
         // LetterNumber can have a uppercase form (e.g. \u2172 'iii' letter which will be 'III')
         foreach (char c in GetTestCharsNotInCategory(UnicodeCategory.LowercaseLetter, UnicodeCategory.TitlecaseLetter, UnicodeCategory.LetterNumber))
         {
-            char lc = char.ToUpperInvariant(c);
-            Assert.Equal(c, lc);
+            Assert.Equal(c, char.ToUpperInvariant(c));
         }
     }
 
-    private static void ValidateTryParse(string s, char expected, bool shouldSucceed)
+
+    [Theory]
+    [InlineData("a", 'a')]
+    [InlineData("4", '4')]
+    [InlineData(" ", ' ')]
+    [InlineData("\n", '\n')]
+    [InlineData("\0", '\0')]
+    [InlineData("\u0135", '\u0135')]
+    [InlineData("\u05d9", '\u05d9')]
+    [InlineData("\ue001", '\ue001')] // Private use codepoint
+    public static void TestParse(string s, char expected)
     {
-        char result;
-        Assert.Equal(shouldSucceed, char.TryParse(s, out result));
-        if (shouldSucceed)
-            Assert.Equal(expected, result);
+        VerifyParse(s, expected);
     }
 
     [Fact]
-    public static void TestTryParse()
-    {
-        // Boolean Char.TryParse(String, Char)
-        ValidateTryParse("a", 'a', true);
-        ValidateTryParse("4", '4', true);
-        ValidateTryParse(" ", ' ', true);
-        ValidateTryParse("\n", '\n', true);
-        ValidateTryParse("\0", '\0', true);
-        ValidateTryParse("\u0135", '\u0135', true);
-        ValidateTryParse("\u05d9", '\u05d9', true);
-        ValidateTryParse("\ud801", '\ud801', true);  // high surrogate
-        ValidateTryParse("\udc01", '\udc01', true);  // low surrogate
-        ValidateTryParse("\ue001", '\ue001', true);  // private use codepoint
-
-        // Fail cases
-        ValidateTryParse(null, '\0', false);
-        ValidateTryParse("", '\0', false);
-        ValidateTryParse("\n\r", '\0', false);
-        ValidateTryParse("kj", '\0', false);
-        ValidateTryParse(" a", '\0', false);
-        ValidateTryParse("a ", '\0', false);
-        ValidateTryParse("\\u0135", '\0', false);
-        ValidateTryParse("\u01356", '\0', false);
-        ValidateTryParse("\ud801\udc01", '\0', false);   // surrogate pair
+    public static void TestParse_Surrogate()
+    {
```

Comment:
Why are these separated out from TestParse?  Again because of xunit?  If so, as noted please add a comment with the issue number so these can be recombined later.